src/components/Settings/Settings.tsx

```diff
@@ -235,49 +123,48 @@ export class Settings extends Component<Props, State> {
      * @param resourceType  ResourceTypes, the type of the control that changed
      */
     handleToggleChange = (isOn: boolean, resourceType: ResourceTypes) => {
-        let { cameraOn, audioOn } = this.state;
+        const { cameraOn, audioOn } = this.state;
+
+        let newCameraOn = resourceType === ResourceTypes.VIDEO ? isOn : cameraOn;
+        let newAudioOn = resourceType === ResourceTypes.AUDIO ? isOn : audioOn;
 
         // do not allow both camera and audio to be turned off
-        cameraOn = !isOn && resourceType === ResourceTypes.AUDIO ? true : cameraOn;
-        audioOn = !isOn && resourceType === ResourceTypes.VIDEO ? true : audioOn;
+        newCameraOn = !isOn && resourceType === ResourceTypes.AUDIO ? true : newCameraOn;
+        newAudioOn = !isOn && resourceType === ResourceTypes.VIDEO ? true : newAudioOn;
 
-        this.setState(
-            (prevState: State) => {
-                return {
-                    cameraOn: resourceType === ResourceTypes.VIDEO ? isOn : cameraOn,
-                    audioOn: resourceType === ResourceTypes.AUDIO ? isOn : audioOn,
-                    screenOn:
-                        resourceType === ResourceTypes.SCREEN_SHARE ? isOn : prevState.screenOn
-                };
-            },
-            () => {
-                this.saveSettings();
-            }
-        );
-    };
+        this.setState((prevState: State) => {
+            return {
+                cameraOn: newCameraOn,
+                audioOn: newAudioOn,
+                screenOn: resourceType === ResourceTypes.SCREEN_SHARE ? isOn : prevState.screenOn
+            };
+        });
 
-    saveSettings = () => {
```

Comment:
since the settings have been changed, I don't know why you have removed this function.
is does exactly what is supposed to do.
any time you want to post to `this.props.onSettingsChanged`, you can just fire this function. 
I suggest to bring it back

Reply:
I did not remove the logic - just united 2 functions into handle setting change